symengine/basic.h

```diff
@@ -97,7 +97,15 @@ class Basic : public EnableRCPFromThis<Basic>
     mutable hash_t hash_; // This holds the hash value
 #endif // WITH_SYMENGINE_THREAD_SAFE
 public:
+#ifdef USE_VIRTUAL_TYPEID
     virtual TypeID get_type_code() const = 0;
+#else
+    TypeID type_code_;
+    inline TypeID get_type_code() const
+    {
+        return type_code_;
```

Comment:
To test something out, I completely removed the code segments corresponding to the previous implementation (ie with the virtual getters). I removed everything inside the`#ifdef USE_VIRTUAL_TYPEID` and kept only the `#else` from `basic.h` (same as putting a `#undef USE_VIRTUAL_TYPEID` in the file, above the code?)

I then run `g++ -E -std=c++11 symengine/basic.cpp  > raw` to see what it looks like after preprocessing, and I see that after preprocessing the `class Basic` has the method `virtual TypeID get_type_code() const = 0;` even though it doesn't not even exist in the file! anywhere!

Can someone explain what is happening?